encryption.tex

```diff
@@ -0,0 +1,202 @@
+
+
+\subsection{Clevis Puppet Profile}
+\begin{lstlisting}[language=Java]
+#Clevis Profile
+class profile::core::clevis() {
+  $packages = [
+    'clevis',
+    'clevis-luks',
+    'clevis-dracut'
+  ]
+
+  ##Add require packages
+  package { $packages:
+    ensure => 'present',
+  }
+  ->exec { '/sbin/dracut -f --regenerate-all':
+    path   => ['/usr/bin', '/sbin'],
+    onlyif => 'test ! -f /usr/lib/dracut/modules.d/60clevis/clevis-hook.sh'
+  }
+}
+\end{lstlisting}
+
+This profile install the clevis packages needed to encrypt and manage the LUKS encryption drives. This is not quite required, cause the clevis packages are being installed during provisioning, but, it grant some useful tools like 'cryptosetup' to check the subscribed Tang servers.
+
+\newpage
+\section{Tang Server - Decryption Server}
+
+\subsection{Tang Service}
+
+Tang \footnote[3]{https://github.com/latchset/tang} is a server for binding data to network presence. In simple terms: you have some data, but you only want it to be available when the system containing the data is on a certain, usually secure, network, This is where Tang comes in.
+First the client gets a list of the Tang server's advertised asymmetric keys. This can happen online by a simple HTTP GET. 
+Second, the client uses one of these public keys to generate a unique, cryptographically strong encryption key. The data is then encrypted using this key. Once the data us encrypted, the key is discarded. Some small metadata is produced as part of this operation which the client should store in a convenient location. This process of encrypting data is the provisioning step.
+Third, when the client is ready to access its data, it simply loads the metadata produced in the provisioning step and performs an HTTP POST in order to recover the encryption key. This process of encrypting data is the provisioning step.
+
+\vskip 2cm
+\begin{figure}
+  \includegraphics[width=14cm]{images/image3.png}
+  \centering
+  \caption{LUKS device interaction with Tang Server}
+\end{figure}
+
+\newpage
+\subsection{Tang Puppet Profile and Role}
+
+\begin{lstlisting}[language=Java]
+#Tang Profile
+class profile::core::tang() {
+  #Variables
+  $packages = ['tang']
+  #Add require packages
+  package { $packages:
+    ensure => 'present',
+  }
+  #Ensure tang service is running
+  service { 'tangd.socket':
+    ensure  => 'running',
+    require => Package[$packages],
+  }
+  #Create tangd.socket override directory
+  file { 'tangd.socket.d':
+    ensure => 'directory',
+    path   => '/etc/systemd/system',
+    mode   => '0755',
+  }
+  #Create override config file
+  file { 'override.conf':
+    ensure  => 'present',
+    path    => '/etc/systemd/system/tangd.socket.d',
+    mode    => '0644',
+    require => File['tangd.socket.d'],
+    content => @(OVERRIDE/L)
+      [Socket]
+      ListenStream=
+      ListenStream=7500
+      | OVERRIDE
+  }
+  ->exec { '/usr/bin/systemctl daemon-reload':
```

Comment:
The `service` type should be smart enough to automatically trigger `daemon-reload`.  

https://github.com/puppetlabs/puppet/blob/main/lib/puppet/provider/service/systemd.rb#L141-L154

If it isn't doing so, please add a comment saying that it isn't working and file a bug report.

Reply:
I'll notify as soon as tested with the drop-in module

Reply:
worked with the systemd module!